src/services/resource.js

```diff
@@ -210,6 +594,208 @@ module.exports = class resourceHelper {
+				})
+			}
+		})
+		const resourceFilter = {
+			[Op.or]: resourceTypeStagesConfig,
+			organization_id,
+			status: { [Op.in]: [common.RESOURCE_STATUS_SUBMITTED, common.RESOURCE_STATUS_IN_REVIEW] },
+		}
+		const resourcesDetails = await resourceQueries.findAll(resourceFilter, ['id'])
+		let resoureId = []
+		if (resourcesDetails) {
+			resoureId = resourcesDetails.map((item) => {
+				return item.id
+			})
+		}
+		return resoureId
+	}
+
+	static async findParallelResources(organization_id, resourceTypes = []) {
+		const resourceFilter = {
+			type: {
+				[Op.in]: resourceTypes,
+			},
+			organization_id,
+			status: { [Op.in]: [common.RESOURCE_STATUS_SUBMITTED, common.RESOURCE_STATUS_IN_REVIEW] },
+		}
+		let resoureId = []
+		const resourcesDetails = await resourceQueries.findAll(resourceFilter, ['id'])
+		if (resourcesDetails) {
+			resoureId = resourcesDetails.map((item) => {
+				return item.id
+			})
+		}
+		return resoureId
+	}
+	// fetch all the resource ids from reviews table which is assigned to the reviewer
+	//  and already started reviewer
+	static async findResourcesAssignedToReviewer(organization_ids, logged_in_user_id) {
+		const reviewsFilter = {
+			organization_id: { [Op.in]: organization_ids },
+			reviewer_id: logged_in_user_id,
+			status: { [Op.in]: common.REVIEW_STATUS_UP_FOR_REVIEW },
+		}
+
+		let res = []
+
+		const reviewsResponse = await reviewsQueries.findAll(reviewsFilter, ['resource_id'])
+		if (reviewsResponse.length > 0) {
+			res = reviewsResponse.map((item) => {
+				return item.resource_id
+			})
+		}
+		return res
+	}
+
+	static async findResourcesPickedUpByAnotherReviewer(loggedInUserId, openToAllResourcesMatchingMyLevel) {
+		// remove all the resouces in sequential review picked up by another reviewer
+		const reviewsFilter = {
+			resource_id: { [Op.in]: openToAllResourcesMatchingMyLevel },
+			status: {
+				[Op.in]: [
+					common.REVIEW_STATUS_INPROGRESS,
+					common.REVIEW_STATUS_CHANGES_UPDATED,
+					common.REVIEW_STATUS_REQUESTED_FOR_CHANGES,
+					common.RESOURCE_STATUS_STARTED,
+				],
+			},
+			reviewer_id: { [Op.notIn]: [loggedInUserId] },
+		}
+		const reviewsResponse = await reviewsQueries.findAll(reviewsFilter, ['resource_id'])
+		let resourceIdsToBeRemoved = []
+		if (reviewsResponse) {
+			// push resource ids to resourceIdsToBeRemoved array
+			resourceIdsToBeRemoved = reviewsResponse.map((item) => item.resource_id)
+			return resourceIdsToBeRemoved
+		}
+		return resourceIdsToBeRemoved
+	}
+
+	// this function fetches review levels from the reviews table for
+	// an organization, roles and resouce type and returns this as an object
+	// Ex : this function returns
+	// {
+	// 	project : 1,
+	// 	observation : 4
+	// }
+	static async fetchResourceReviewLevel(organization_id, userRoleTitles, resourceTypeList) {
+		// fetch review levels according to roles in the organization
+		const reviewLevelDetails = await reviewStagesQueries.findAll(
```

Comment:
@adithyadinesh0412  what if the review type i sequential and reviewLevelDetails is empty